tempesta_fw/work_queue.c

```diff
@@ -139,6 +139,10 @@ __tfw_wq_push(TfwRBQueue *q, void *ptr)
 			if (head >= atomic64_read(&q->last_tail) + QSZ) {
 				/* The queue is full, don't wait consumers. */
 				atomic64_set(&pos->head, LLONG_MAX);
+				if (sync) {
+					cpu_relax();
+					goto retry;
+				}
```

Comment:
No, actually the branch is for function returning. We don't need to do second try. If `sync == true`, then just call `__update_guards()`. I think we don't need to call `cpu_relax()` since there are relatively many code to execute.

So the adjustment could be just:

        if (!sync && head >= atomic64_read(&q->last_tail) + QSZ) {